src/coreclr/pal/src/CMakeLists.txt

```diff
@@ -76,7 +76,6 @@ endif(FEATURE_ENABLE_NO_ADDRESS_SPACE_RANDOMIZATION)
 add_definitions(-DLP64COMPATIBLE)
 add_definitions(-DCORECLR)
 add_definitions(-DPIC)
-add_definitions(-D_FILE_OFFSET_BITS=64)
```

Comment:
@am11 why have you removed this line and the same one in the configure.cmake and inb the hosts CMakeLists? It is not correct, we want to support 64 bit file offsets on all platforms. The PAL tests on ARM are now failing due to this (see #58481). I don't see this removal mentioned in the PR description. I have not noticed this before, I was OOF when this PR was created and merged.

Reply:
@janvorli, it was moved at common place: https://github.com/dotnet/runtime/blob/5181e1cd5b3fde9da15fe3f8d3eceaa9d3f64e01/eng/native/configurecompiler.cmake#L280-L281

Reply:
But that doesn't influence the compiler options in the configure.cmake. So while this place and the hosts are likely ok, the configure.cmake change causes the problem, as we detect off_t as 4 bytes due to the missing define.

Reply:
I think you can double-check, but I am pretty sure global defines are used in cmake introspection unless explicitly stripped off.

Besides, #58481 was opened 13 hours before this PR was merged..

Reply:
Ah I mixed up; was checking on mobile, but it does seem like first occurrence of `File sizes do not match up.` happened after this was merged. The one on 24th of August is different error `Call to SetEndOfFile failed with 112`. Will check it out shortly.

Reply:
I've just verified that adding that line back to configure.cmake fixes the failing tests (I've tried without it to eliminate the possibility that something else is in play here). And the global defines are really not used in the introspection, I've checked the CMakeOutput.txt and the only definitions passed to the compiler were the ones in the CMAKE_REQUIRED_DEFINITIONS:
```
Run Build Command(s):/usr/bin/make cmTC_087a6/fast && /usr/bin/make -f CMakeFiles/cmTC_087a6.dir/build.make CMakeFiles/cmTC_087a6.dir/build
make[1]: Entering directory '/home/janvorli/git/runtime/artifacts/obj/coreclr/Linux.arm.Debug/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_087a6.dir/SIZEOF_OFF_T.c.o
/usr/bin/clang-9 -fPIE   -D_BSD_SOURCE -D_SVID_SOURCE -D_DEFAULT_SOURCE -D_POSIX_C_SOURCE=200809L -o CMakeFiles/cmTC_087a6.dir/SIZEOF_OFF_T.c.o   -c /home/janvorli/git/runtime/artifacts/obj/coreclr/Linux.arm.Debug/CMakeFiles/CheckTypeSize/SIZEOF_OFF_T.c
Linking C executable cmTC_087a6
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_087a6.dir/link.txt --verbose=1
/usr/bin/clang-9     -pie -Wl,--build-id=sha1 -Wl,-z,relro,-z,now  CMakeFiles/cmTC_087a6.dir/SIZEOF_OFF_T.c.o  -o cmTC_087a6
make[1]: Leaving directory '/home/janvorli/git/runtime/artifacts/obj/coreclr/Linux.arm.Debug/CMakeFiles/CMakeTmp'

```

Reply:
Absolutely, we `CMAKE_REQUIRED_DEFINITIONS` so whatever it had before is reset. https://github.com/dotnet/runtime/pull/59504 for the fix. Thanks for spotting it! :)